src/map/clan.c

```diff
@@ -0,0 +1,792 @@
+					continue;
+				}
+
+				if (!libconfig->setting_lookup_string(clan_, "Name", &aName)) {
+					ShowError("clan_config_read: failed to find 'Name' for clan #%d\n", i);
+					continue;
+				}
+
+				if (!libconfig->setting_lookup_string(clan_, "Leader", &aLeader)) {
+					ShowError("clan_config_read: failed to find 'Leader' for clan #%d\n", i);
+					continue;
+				}
+
+				if (!libconfig->setting_lookup_string(clan_, "Map", &aMap)) {
+					ShowError("clan_config_read: failed to find 'Map' for clan #%d\n", i);
+					continue;
+				}
+
+				CREATE(c, struct clan, 1);
+
+				c->clan_id = id;
+				safestrncpy(c->constant, aConst, NAME_LENGTH);
+				safestrncpy(c->name, aName, NAME_LENGTH);
+				safestrncpy(c->master, aLeader, NAME_LENGTH);
+				safestrncpy(c->map, aMap, MAP_NAME_LENGTH_EXT);
+				c->max_member = clan->max;
+				if (c->max_member > MAX_CLAN) {
+					ShowWarning("Clan %d:%s specifies higher capacity (%d) than MAX_CLAN (%d)\n", c->clan_id, c->name, c->max_member, MAX_CLAN);
+					c->max_member = MAX_CLAN;
+				}
+				c->connect_member = 0;
+
+				if (libconfig->setting_lookup_int(clan_, "KickTime", &kicktime)) {
+					c->kick_time = 60 * 60 * kicktime;
+				} else {
+					c->kick_time = global_kicktime;
+				}
+				if (libconfig->setting_lookup_int(clan_, "CheckTime", &kickchecktime)) {
+					c->check_time = 60 * 60 * max(1, kickchecktime) * 1000;
+				} else {
+					c->check_time = global_checktime;
+				}
+
+
+				if ((buff = libconfig->setting_get_member(clan_, "Buff")) != NULL) {
+					struct clan_buff *b = &c->buff;
+					const char *str = NULL;
+
+					if (!libconfig->setting_lookup_int(buff, "Icon", &b->icon)) {
+						const char *temp_str = NULL;
+						if (libconfig->setting_lookup_string(buff, "Icon", &temp_str)) {
+							if (*temp_str) {
+								if (!script->get_constant(temp_str, &b->icon)) {
+									ShowWarning("Clan %d: Invalid Icon Value. Defaulting to SI_BLANK.\n", c->clan_id);
+									b->icon = -1;
+								}
+							}
+						}
+					}
+
+					if (libconfig->setting_lookup_string(buff, "Script", &str)) {
+						b->script = *str ? script->parse(str, config_filename, -b->icon, SCRIPT_IGNORE_EXTERNAL_BRACKETS, NULL) : NULL;
+					}
+				}
+
+				if (allies != NULL) {
+					int a = 0;
+					const char *ally_const;
+
+					for (a = 0; a < libconfig->setting_length(allies) &&  a < clan->maxalliances; a++) {
+						if ((ally_const = libconfig->setting_get_string_elem(allies, a)) != NULL)
+							safestrncpy(c->allies[a].constant, ally_const, NAME_LENGTH);
+					}
+				}
+				if (antagonists != NULL) {
+					int o = 0;
+					const char *antg_const;
+
+					for (o = 0; o < libconfig->setting_length(antagonists) && o < clan->maxalliances; o++) {
+						if ((antg_const = libconfig->setting_get_string_elem(antagonists, o)) != NULL)
+							safestrncpy(c->antagonists[o].constant, antg_const, NAME_LENGTH);
+					}
+				}
+				idb_put(clan->db, c->clan_id, c);
```

Comment:
missing call to function for read additional fields for plugins. See how it works in other dbs